api/src/wfl/module/covid.clj

```diff
@@ -0,0 +1,41 @@
+(ns wfl.module.covid
+  "Handle COVID processing."
+  (:require [clojure.data.json :as json]
+            [clojure.set :as set]
+            [clojure.string :as str]
+            [clojure.tools.logging.readable :as log]
+            [wfl.api.workloads :as workloads :refer [defoverload]]
+            [wfl.jdbc :as jdbc]
+            [wfl.module.batch :as batch]
+            [wfl.service.datarepo :as datarepo]
+            [wfl.service.firecloud :as firecloud]
+            [wfl.service.google.storage :as gcs]
+            [wfl.service.postgres :as postgres]
+            [wfl.util :as util]
+            [wfl.wfl :as wfl]))
+
+(def pipeline "Sarscov2IlluminaFull")
+
+;; TODO: implement COVID workload creation
+;;  - make sure permissions/inputs are right upfront
+;;  - dispatch on source/sink/executor
+;;  - store information into database
+(defn create-covid-workload!
+  [tx {:keys [source sink executor] :as request}])
+
+;; TODO: implement progressive (private) functions inside this update loop
+(defn update-covid-workload!
+  "Use transaction `tx` to batch-update `workload` statuses."
+  [tx {:keys [started finished] :as workload}]
+  (letfn [(update! [{:keys [id] :as workload}]
+            (postgres/batch-update-workflow-statuses! tx workload)
+            (postgres/update-workload-status! tx workload)
+            (workloads/load-workload-for-id tx id))]
```

Comment:
I think I understand what the other functions are doing in this loop but not this one.  Would be a help to hear to explain it!